arXiv.org.js

```diff
@@ -35,217 +35,396 @@
+	if (dts.length !== dds.length) {
+		Z.debug(`Warning: unexpected number of <dt> and <dd> elements: ${dts.length} !== ${dds.length}`);
+	}
+	let length = Math.min(dts.length, dds.length);
+	for (let i = 0; i < length; i++) {
+		let id = text(dts[i], "a[title='Abstract']")
+			.trim()
+			.replace(/^arXiv:/, "");
+		let title = ZU.trimInternal(text(dds[i], ".list-title"))
+			.replace(/^Title:\s*/, "");
+		if (!id || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[id] = title;
 	}
+	return found && items;
 }
 
-function doWeb(doc, url) {
+async function doWeb(doc, url) {
 	if (detectWeb(doc, url) == 'multiple') {
-		var rows = ZU.xpath(doc, '//div[@id="dlpage"]/dl/dt');
-		var getTitleId;
-		if (rows.length) {
-			// arXiv.org format
-			getTitleId = function (row) {
-				var id = ZU.xpathText(row, './/a[@title="Abstract"]').trim().substr(6); // Trim off arXiv:
-				var title = ZU.trimInternal(
-					ZU.xpathText(row, './following-sibling::dd[1]//div[contains(@class, "list-title")]/text()[last()]'));
-				return {
-					title: title,
-					id: id
-				};
-			};
-		}
-		else if ((rows = ZU.xpath(doc, '//table/tbody/tr[./td[@class="lti"]]')).length) {
-			// eprintweb.org format
-			getTitleId = function (row) {
-				var title = ZU.trimInternal(ZU.xpathText(row, './td'));
-				var id = ZU.xpathText(row, './following-sibling::tr[.//a][1]/td/b').trim().substr(6);
-				return {
-					title: title,
-					id: id
-				};
-			};
-		}
-		else {
-			throw new Error("Unrecognized multiples format");
-		}
+		var items = getSearchResults(doc);
 		
-		var items = {};
-		for (let i = 0; i < rows.length; i++) {
-			var row = getTitleId(rows[i]);
-			items[row.id] = row.title;
+		let selectedItems = await Z.selectItems(items);
+		if (selectedItems) {
+			let apiURL = `https://export.arxiv.org/api/query?id_list=${encodeURIComponent(Object.keys(selectedItems).join(','))}`;
+			let document = await requestDocument(apiURL);
+			parseAtom(document);
 		}
-		
-		Z.selectItems(items, function (items) {
-			if (!items) return;
-			
-			var urls = [];
-			for (var id in items) {
-				urls.push('https://export.arxiv.org/oai2'
-					+ '?verb=GetRecord&metadataPrefix=oai_dc'
-					+ '&identifier=oai%3AarXiv.org%3A' + encodeURIComponent(id)
-				);
-			}
-			
-			ZU.doGet(urls, parseXML);
-		});
 	}
 	else {
-		var id;
-		var versionMatch = url.match(/v(\d+)(\.pdf)?([?#].+)?$/);
+		let id = url.match(/(?:pdf|abs)\/(.+)(?:\.pdf)?/)[1];
+		let versionMatch = url.match(/v(\d+)(\.pdf)?([?#].+)?$/);
 		if (versionMatch) {
 			version = versionMatch[1];
 		}
 		arxivDOI = text(doc, '.arxivdoi>a');
-		var p = url.indexOf("/pdf/");
-		if (p > -1) {
-			id = url.substring(p + 5, url.length - 4);
-		}
-		else {
-			id = ZU.xpathText(doc, '(//span[@class="arxivid"]/a)[1]')
-				|| ZU.xpathText(doc, '//b[starts-with(normalize-space(text()),"arXiv:")]');
+
+		if (!id) { // Honestly not sure where this might still be needed
```

Comment:
The Atom endpoint handles version numbers well, so I don't see cases where we wouldn't rely on the url to get the ID. Might want to delete this.